draft-ietf-httpbis-origin-frame.md

```diff
@@ -70,80 +75,183 @@ The ORIGIN HTTP/2 frame ({{!RFC7540}}, Section 4) allows a server to indicate wh
 frame to indicate multiple origins:
 
 Origin-Len:
-: An unsigned, 16-bit integer indicating the length, in octets, of the Origin field.
+: An unsigned, 16-bit integer indicating the length, in octets, of the ASCII-Origin field.
 
 Origin:
 : An optional sequence of characters containing the ASCII serialization of an origin ({{!RFC6454}}, Section 6.2) that the sender believes this connection is or could be authoritative for.
 
-The ORIGIN frame defines the following flags:
+The ORIGIN frame does not define any flags. However, future updates to this specification MAY
+define flags. See {{process}}.
+
+
+## Processing ORIGIN Frames {#process}
+
+The ORIGIN frame is a non-critical extension to HTTP/2. Endpoints that do not support this frame
+can safely ignore it upon receipt.
+
+When received by an implementing client, it is used to initialise and manipulate the Origin Set
+(see {{set}}), thereby changing how the client establishes authority for origin servers (see
+{{authority}}).
+
+The origin frame MUST be sent on stream 0; an ORIGIN frame on any other stream is invalid and MUST
+be ignored.
 
-CLEAR (0x1):
-: Indicates that the Origin Set MUST be reset to an empty set before processing the contents of the frame it occurs upon.
+Likewise, the ORIGIN frame is only valid on connections with the "h2" protocol identifier, or when
+specifically nominated by the protocol's definition; it MUST be ignored when received on a
+connection with the "h2c" protocol identifier.
 
-REMOVE (0x2): 
-: Indicates that the origin(s) carried in the payload must be removed from the Origin Set, if present; if not present, it/they have no effect.
+This specification does not define any flags for the ORIGIN frame, but future updates might use
+them to change its semantics. The first four flags (0x1, 0x2, 0x4 and 0x8) are reserved for
+backwards-incompatible changes, and therefore when any of them are set, the ORIGIN frame containing
+them MUST be ignored by clients conforming to this specification. The remaining flags are reserved
+for backwards-compatible changes, and do not affect processing by clients conformant to this
+specification.
+
+The ORIGIN frame is processed hop-by-hop. An intermediary MUST NOT forward ORIGIN frames. Clients
+configured to use a proxy MUST ignore any ORIGIN frames received from it.
+
+Each ASCII-Origin field in the frame's payload MUST be parsed as an ASCII serialisation of an
+origin ({{!RFC6454}}, Section 6.2). If parsing fails, the field MUST be ignored.
+
+Senders should note that, as per {{!RFC6454}} Section 4, the values in an ORIGIN header need to be
+case-normalised before serialisation.
+
+Once parsed, the value MUST have:
+
+* a scheme of "https", 
+* a host that is reflected in a `subjectAltName` of the connection's TLS certificate (using the wildcard rules defined in {{!RFC2818}}, Section 3.1), and 
+* a port that reflects the connection's remote port on the client. 
```

Comment:
@mcmanus see latest commit; I defer the cert checks until use of the origin, and I think that takes care of the concerns here.